src/main/java/fi/oph/akt/repository/LanguagePairRepository.java

```diff
@@ -15,4 +15,10 @@
 			+ " JOIN lp.authorisation a JOIN a.translator t WHERE lp.permissionToPublish=true AND t.id IN ?1")
 	List<TranslatorLanguagePairProjection> findTranslatorLanguagePairsForPublicListing(Iterable<Long> translatorIds);
 
+	@Query("SELECT DISTINCT lp.fromLang FROM LanguagePair lp ORDER BY lp.fromLang")
+	List<String> getDistinctFromLangs();
+
+	@Query("SELECT DISTINCT lp.toLang FROM LanguagePair lp ORDER BY lp.toLang")
+	List<String> getDistinctToLangs();
```

Comment:
Kannattaisiko näissä kuitenkin olla permissionToPublish=true, koska: kääntäjissä on vain ne joilla on kielipareissa julkaisu sallittu. Jos näissä ei haeta vain permissionToPublish=true niin kielivalintalistassa on kieliä joille ei ikinä löydy ketään. Jos siis se yksinäinen esperanton kääntäjä ei halua tietojaan julki, näkyy esperanto kielivalinnassa ihan turhaan. Ei ole kyllä iso asia tod näk.

Reply:
Mietin tätä itsekin, mutta alustavasti ajattelin, että noita tuskin on kannassa montaa jos useimmiten yhtäkään. Virkailijalistauksen yhteydessä käytetään todennäköisesti samaa logiikkaa kielilistojen hakemiseen ja mietin, olisko näitä tietokantahakumetodeja mitään järkeä yhdistää tyyliin

```
	@Query("SELECT DISTINCT lp.fromLang FROM LanguagePair lp WHERE lp.permissionToPublish IN (true, ?1) ORDER BY lp.fromLang")
	List<String> getDistinctFromLangs(boolean requirePermissionToPublish);
```

Samanlainen tapa olisi mahdollista myös metodien `findTranslatorLanguagePairs` ja `findTranslatorLanguagePairsForPublicListing` kanssa. Tuo tosin tuntuu typerältä tarkistaa, onko boolean-tyyppisen sarakkeen arvo joukossa `(true, true)` tai `(true, false)`.

Reply:
Tai sitten tarpeen vaatiessa luo "duplikaattimetodit" myös tähän tarkoitukseen virkailijanäkymän kielilistauksille erikseen.

Reply:
Aika teoreettinenhan tuo taitaa olla.Ehkä ei nyt tehdä virkailijanäkymää varten mitään, muutetaan sitten kun se tehdään jos on tarpeen?

Reply:
Jos teen tiketin tuosta tarkistuksesta ja lisätään myöhemmin jos tarpeen? Jos ymmärsin oikein että mietit sen tarvetta aika teoreettisena.

Reply:
Juu lisätään myöhemmin jos tarpeen.

Reply:
OPHAKTKEH-121 luotu